gpii/node_modules/processReporter/src/processesBridge.js

```diff
@@ -0,0 +1,236 @@
+            args: ["{arguments}.0", "{arguments}.1"]
+            // command, procArray (optional)
+        },
+        findFirstProcessByCommand: {
+            funcName: "gpii.processes.findFirstProcessByCommand",
+            args: ["{arguments}.0", "{arguments}.1"]
+            // command, procArray (optional)
+        },
+        isRunning: {
+            funcName: "gpii.processes.isRunning",
+            args: ["{arguments}.0"]   // state (string)
+        },
+        updateProcInfo: {
+            funcName: "gpii.processes.updateProcInfo",
+            args: ["{arguments}.0"]   // process info structure
+        },
+        initProcInfoNotRunning: {
+            funcName: "gpii.processes.initProcInfoNotRunning",
+            args: ["{arguments}.0"]   // command name (string)
+        }
+    }
+});
+
+/**
+ * Return a list of process information objects corresponding to the names
+ * of each of the passed in commands.  If nothing is found, returns an empty
+ * array.
+ *
+ * @param commandNames {Array} - the names of the processes to inspect.
+ * @return {Array} - Array of procInfo objects for each command name.  Empty
+ *                   if no processes are found.
+ */
+gpii.processes.findSolutionsByCommands = function (commandNames) {
+    if (!!commandNames) {
+        return fluid.accumulate(commandNames, function (aCommand, matches) {
+            var procInfos = gpii.processes.findProcessesByCommand(aCommand);
+            matches = matches.concat(procInfos);
+            return matches;
+        }, []);
+    }
+    else {
+        return [];
+    }
+};
+
+/**
+ * Return an list of process information objects corresponding to the
+ * pids (process id numbers) passed in.  If nothing is found, returns an empty
+ * array.
+ *
+ * @param pids {Array} - the process ids of the processes to inspect.
+ * @return {Array} - Array of procInfo objects for each pid.  Empty
+ *                   if no corresponding processes are found.
+ */
+gpii.processes.findSolutionsByPids = function (pids) {
+    if (!!pids) {
+        return fluid.accumulate(pids, function (aPid, matches) {
+            var found = gpii.processes.findProcessByPid(aPid);
+            if (found !== null) {
+                matches.push(found);
+            }
+            return matches;
+        }, []);
+    }
+    else {
+        return [];
+    }
+};
+
+/**
+ * Return a process information object corresponding to THE given process
+ * id (process id number).  Returns null if there is no such process.
+ *
+ * @param pid {Number} - the process id of the processe to inspect.
+ * @param procArray {Array} Optional - an array of process information
+ *                                     objects to search.
+ * @return {Object} - A proccess information object for the process with the
+ *                    given id.  Returns null if there is no such process.
+ */
+gpii.processes.findProcessByPid = function (pid, procArray) {
```

Comment:
@amb26, Do you mean by passing a "{that}" argument?  That is, the declaration of the arguments for each invoker should pass "{that}"?